Update ports rbenv and ruby-build: Break rbenv dependency from ruby-build

halostatue opened the pull request:
#### Description

rbenv: update to version 1.2.0

Add a notes block advising the installation of port ruby-build if the
subcommand `rbenv install` is desired.

Closes: https://trac.macports.org/ticket/65329
Signed-off-by: Austin Ziegler <austin@zieglers.ca>

ruby-build: remove incorrect dependency

- Removes the dependency on `rbenv`, as ruby-build does not require
  rbenv to work (but rbenv can integrate ruby-build as a plug-in).

- Add a notes block that this port no longer installs port rbenv
  automatically, and advising manual installation of port rbenv.

Closes: https://trac.macports.org/ticket/65329
Closes: https://github.com/macports/macports-ports/pulls/16273
Closes: https://github.com/macports/macports-ports/pulls/16274
Signed-off-by: Austin Ziegler <austin@zieglers.ca>

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.6 21G115 arm64
Xcode 13.4.1 13F100

###### Verification

Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

halostatue commented:
This is a potentially *breaking* fix to `ruby-build` and `rbenv` in that the dependency is broken between `ruby-build` to `rbenv`  and added as an *optional* (variant-based) dependency *from* `rbenv` to `ruby-build`. The "safer" fix (but arguably still incorrect because the default behaviour still installs `rbenv`) is #16273.

halostatue commented:
I split this version into two commits.

amake commented on ruby/ruby-build/Portfile:
```diff
@@ -3,24 +3,23 @@
 PortGroup           github 1.0
 
-github.setup        rbenv ruby-build 20220713 v
+github.setup        rbenv ruby-build 20221225 v
+revision            1
```
Since you are bumping the version, you don't need to bump the revision.

halostatue replied:
I had bumped the revision here because this is a *change* to the way that this port works. I‘m not sure what sort of signal that would require otherwise. I can change it back.

amake replied:
Thanks for clarifying, but that's not what `revision` means in MacPorts. Bumping `revision` is for when `version` is *un*changed but something else about that port has changed (dependencies, installed files, default variants, etc.) such that the port needs to be rebuilt. It is expected to reset `revision` to 0 when bumping the `version`.

amake replied:
Now that #17090 has been merged and this PR will no longer bump the `version`, you _should_ bump the `revision` due to the change in dependencies.

halostatue commented:
Updated so that there’s no conflict. I also added Closes lines for both this PR and the alternate PR (#16273) to both this and the other, since only *one* of the two should be chosen. I think that this is the *better* of the two fixes, but it changes the (optional) dependency direction (correctly).

amake commented on ruby/rbenv/files/patch-libexec-rbenv-init.diff:
```diff
@@ -0,0 +1,27 @@
+This patch is needed to make
+    eval "$(rbenv init -)"
+work without complaining about
+    -bash: /opt/local/bin/../completions/rbenv.bash: No such file or directory
```
This patch isn't actually being used by the port—you need to specify `patchfiles patch-libexec-rbenv-init.diff`.

Also I don't see the error you describe. Could you provide instructions for reproducing the issue?

halostatue replied:
This patch was previously included. It may not be necessary any longer. I will need to check.

amake commented on ruby/rbenv/Portfile:
```diff
@@ -4,9 +4,10 @@ PortSystem          1.0
+revision            1
 license             MIT
 categories          ruby
-platforms           any
+platforms           darwin
```
Is there a reason for this change? Since rbenv is just a bash script, I believe it should run on other platforms as well.

halostatue replied:
The change here was related to `port lint`. I have started following more MacPorts messages, and it seems that `port lint` has not been updated to recognize `any` as a valid value, although it is the correct choice. I will revert it (I will resolve after I have made the change).

amake commented:
I think this approach makes more sense than #16273, but I would suggest one of the following changes:

- I think users of rbenv will probably usually want ruby-build, so the `rubybuild` variant should be a default variant
- Or, alternatively, you could remove the variant entirely and add a `notes` block to rbenv advertising ruby-build

halostatue commented:
I’ll look at the suggested changes and try to get them in place over the next day or so. I don’t use `rbenv`, preferring `chruby` (and now I mostly don’t even use that because I have a `chruby`-like `direnv` handler function). It’s also somewhat difficult because I now use fish instead of bash or zsh, so…

I think that what’s better will be to switch to adding a `notes` block suggesting the installation of `ruby-build` to get the plug-in capability. Should we also add a `notes` block to `ruby-build` so that when someone upgrades to this revision they will see that there is no longer an automatic connection to `rbenv`?

halostatue commented:
I believe that I have resolved most of the issues. I have reworded the commit messages and updated the description of the PR as well.

amake commented:
I am happy with this, but since it's a significant change (and may go against the intention of the original port authors) I think we should wait for @mojca.

halostatue commented:
Also updated ruby-build to version `20230208`.

I’m wondering if there’s any way to get action on this PR and its associated ticket, because the original dependency between `ruby-build` and `rbenv` was incorrect (if anything, the dependency should have been *reversed* as most people that I know who use `rbenv` do so with `ruby-build`, whereas `ruby-build` is one of two general Ruby installers). I’ve subscribed to release notifications for `ruby-build`, so I’m also happy to become a maintainer for this port.

Please note that I have closed #16273 as I don’t believe that is the correct fix, and do not wish to keep updating it for new `ruby-build` releases.

amake commented:
I think this indisputably the right approach and given the maintainer's inaction I will just go ahead and merge it.
